mevolib/cluster/Genes.py

```diff
@@ -1,76 +1,36 @@
 #
-#   MEvoLib  Copyright (C) 2016  J. Alvarez-Jarreta
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
 #
-#   This program comes with ABSOLUTELY NO WARRANTY; for details type `show w'.
-#   This is free software, and you are welcome to redistribute it under certain
-#   conditions; type `show c' for details.
+#     http://www.apache.org/licenses/LICENSE-2.0
 #
-#-------------------------------------------------------------------------------
-# File :  Genes.py
-# Last version :  v1.21 ( 19/Jul/2016 )
-# Description :  Clustering where each resulting set is composed by a gene of
-#       all the input sequences (from GenBank data or reference sequence).
-#-------------------------------------------------------------------------------
-# Historical report :
-#
-#   DATE :  19/Jul/2016
-#   VERSION :  v1.21
-#   AUTHOR(s) :  J. Alvarez-Jarreta
-#   CHANGES :  * Minor bug fixed.
-#
-#   DATE :  08/Jul/2016
-#   VERSION :  v1.20
-#   AUTHOR(s) :  J. Alvarez-Jarreta
-#   CHANGES :  * Fixed an important error in the interpretation and
-#                 implementation of the statistical sampling equation in
-#                 map_seqs() method.
-#
-#   DATE :  15/Jun/2016
-#   VERSION :  v1.11
-#   AUTHOR(s) :  J. Alvarez-Jarreta
-#   CHANGES :  * Fixed two bugs in map_seqs() method: i) an empty dictionary was
-#                  returned if the input file was not in 'genbank' format;
-#                  ii) the error calculation was raising an exception if the
-#                  number of sequences was lower than the sampling size.
-#
-#   DATE :  11/May/2016
-#   VERSION :  v1.10
-#   AUTHOR(s) :  J. Alvarez-Jarreta
-#   CHANGES :  * The map_seqs() method now generates a report of possible wrong
-#                  pairs of terms (maybe a typo in GenBank information?) based
-#                  on sampling statistics applied to the input sequences.
-#              * Fixed a bug in map_seqs() method with a possible random
-#                  ordering of terms for the same gene making it not unique in
-#                  the generation of the qualifier id.
-#
-#   DATE :  07/Feb/2016
-#   VERSION :  v1.00
-#   AUTHOR(s) :  J. Alvarez-Jarreta
-#
-#-------------------------------------------------------------------------------
-
-from __future__ import absolute_import
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Clustering where each resulting set is composed by a gene of all the input sequences 
+(from GenBank data or reference sequence)."""
 
 import tempfile
 import itertools
 import math
+from typing import List, Optional
```

Comment:
It seems things have improved since the last time I did this and you can use `list` directly, without importing it:
```suggestion
from typing import Optional
```